app/lib/dartdoc/models.dart

```diff
@@ -59,6 +74,23 @@ class DartdocEntry extends Object with _$DartdocEntrySerializerMixin {
   String get contentPrefix => '$packageName/$packageVersion/content/$uuid';
 
   List<int> asBytes() => UTF8.encode(JSON.encode(this.toJson()));
+
+  bool requiresNewRun() {
+    if (isNewer(_semanticDartdocVersion, versions.semanticDartdocVersion)) {
+      return true;
+    } else if (dartdocVersion != versions.dartdocVersion) {
```

Comment:
This one took me a couple of seconds to understand, so maybe add a comment. I understood it as:
* If our dartdoc version is newer, regenerate (checked above)
* If dartdoc versions are different, *our version must be older* - so skip generating
* Otherwise, dartdoc versions are the same, so we only regenerate if flutter or customization is newer.

Reply:
That is correct, adding docs.